src/main/kotlin/com/sourcegraph/cody/config/notification/AccountSettingChangeListener.kt

```diff
@@ -25,18 +24,11 @@ class AccountSettingChangeListener(project: Project) : ChangeListener(project) {
             // Notify JCEF about the config changes
             javaToJSBridge?.callJS("pluginSettingsChanged", ConfigUtil.getConfigAsJson(project))
 
-            if (ConfigUtil.isCodyEnabled()) {
-              // Starting the agent is idempotent, so it's OK if we call startAgent multiple times.
-              CodyAgentManager.startAgent(project)
-            } else {
-              // Stopping the agent is idempotent, so it's OK if we call stopAgent multiple times.
-              CodyAgentManager.stopAgent(project)
-            }
-
             // Notify Cody Agent about config changes.
-            val agentServer = CodyAgent.getServer(project)
-            if (ConfigUtil.isCodyEnabled() && agentServer != null) {
-              agentServer.configurationDidChange(ConfigUtil.getAgentConfiguration(project))
+            CodyAgentService.applyAgentOnBackgroundThread(project) { agent ->
+              if (ConfigUtil.isCodyEnabled()) {
```

Comment:
so previously depending on this boolean we were starting or stopping the agent. Is the behavior different now? so is the agent always started?

Reply:
That is something I take liberty to remove but we have to test it. Agent is always started (or if not yet `CodyAgentService.applyAgentOnBackgroundThread` will wait for it to be).
I think starting/stoping should take place when cody settings change, as that is where you can enable/disable cody (and we have proper code  for that in settings).

But if someone see a reason why that code was needed there I will be happy to hear it.